File: Recursions/arraySum.py
def arrSum(arr,si):

    l = len(arr)

    if (si == l):
        return 0


    smallerOutput = arrSum(arr, si + 1)
    return arr[si] + smallerOutput


arr = [1, 2, 3]
print(arrSum(arr, 0))

File: Recursions/arraySum.cpp
#include <bits/stdc++.h>
using namespace std;

void displayArray(int arr[], int n){
    for (int i = 0; i < n; i++){
        cout << arr[i] << ",";
    }
}

int arraySum(int arr[], int n, int si){
    if (si == n){
        return 0;
    }

    int smallerOutput = arraySum(arr, n, si + 1);
    return arr[si] + smallerOutput;
}

int main(){
    int n;
    cin >> n;

    int arr[50];

    for(int i = 0; i < n; i++){
        cin >> arr[i];
    }

    displayArray(arr, n);

    cout << "sum: "<< arraySum(arr, n, 0);
}
